### ISSUE TITLE
[BUG] RuntimeError when passing all nulls to the normalize_characters API

### ISSUE THREAD

**Issue by davidwendt** [rapidsai/cudf#5812]
**Describe the bug**
RuntimeError when passing all nulls to the `normalize_characters` API.
Reference: https://github.com/rapidsai/cudf/issues/5527#issuecomment-666052830

**Steps/Code to reproduce bug**
```
import cudf
cdf = cudf.DataFrame({"A":[None, None,None]})
cdf["A"].astype(str).str.normalize_characters()
   3667         """
   3668         return self._return_or_inplace(
-> 3669             cpp_normalize_characters(self._column, do_lower), **kwargs
   3670         )
   3671 

cudf/_lib/nvtext/normalize.pyx in cudf._lib.nvtext.normalize.normalize_characters()

RuntimeError: cuDF failure at: /opt/conda/envs/rapids/conda-bld/libcudf_1596061003383/work/cpp/include/cudf/detail/utilities/cuda.cuh:64: num_blocks must be > 0
```

**Expected behavior**
Return an empty result and not crash.

**Comment by argenisleon** [comment]
Hi @davidwendt,

Running the same code on nightly `0.16.0a+713.gc152e9b2a`:
```python
import cudf
cdf = cudf.DataFrame({"A":[None, None,None],"B":[None,"1",None],"C":["1","1","1"], "D":["1","1",None]})
cdf["A"].astype(str).str.normalize_characters()
```
Now raise:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    392                         if cls is not object \
    393                                 and callable(cls.__dict__.get('__repr__')):
--> 394                             return _repr_pprint(obj, self, cycle)
    395 
    396             return _default_pprint(obj, self, cycle)

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    698     """A pprint that just redirects to the normal repr function."""
    699     # Find newlines and replace them with p.break_()
--> 700     output = repr(obj)
    701     lines = output.splitlines()
    702     with p.group():

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/cudf/core/series.py in __repr__(self)
   1006             )
   1007         else:
-> 1008             output = preprocess.to_pandas(nullable_pd_dtype=False).__repr__()
   1009 
   1010         lines = output.split("\n")

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/cudf/core/series.py in to_pandas(self, index, **kwargs)
   2012             index = self.index.to_pandas()
   2013         s = self._column.to_pandas(
-> 2014             index=index, nullable_pd_dtype=nullable_pd_dtype
   2015         )
   2016         s.name = self.name

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/cudf/core/column/string.py in to_pandas(self, index, nullable_pd_dtype)
   4620         pd_series = self.to_arrow().to_pandas()
   4621         if index is not None:
-> 4622             pd_series.index = index
   4623         if nullable_pd_dtype:
   4624             return pd_series.astype(pd.StringDtype(), copy=False)

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/pandas/core/generic.py in __setattr__(self, name, value)
   5285         try:
   5286             object.__getattribute__(self, name)
-> 5287             return object.__setattr__(self, name, value)
   5288         except AttributeError:
   5289             pass

pandas/_libs/properties.pyx in pandas._libs.properties.AxisProperty.__set__()

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/pandas/core/series.py in _set_axis(self, axis, labels, fastpath)
    399         object.__setattr__(self, "_index", labels)
    400         if not fastpath:
--> 401             self._data.set_axis(axis, labels)
    402 
    403     def _set_subtyp(self, is_all_dates):

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/pandas/core/internals/managers.py in set_axis(self, axis, new_labels)
    176         if new_len != old_len:
    177             raise ValueError(
--> 178                 f"Length mismatch: Expected axis has {old_len} elements, new "
    179                 f"values have {new_len} elements"
    180             )

ValueError: Length mismatch: Expected axis has 0 elements, new values have 3 elements
```

Applying the funcion on series `B`:
```python
import cudf
cdf = cudf.DataFrame({"A":[None, None,None],"B":[None,"1",None],"C":["1","1","1"], "D":["1","1",None]})
cdf["B"].astype(str).str.normalize_characters()
```

Raise:

```python
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-a5b481588ed4> in <module>
      1 import cudf
      2 cdf = cudf.DataFrame({"A":[None, None,None],"B":[None,"1",None],"C":["1","1","1"], "D":["1","1",None]})
----> 3 cdf["B"].astype(str).str.normalize_characters()

~/anaconda3/envs/rapids_blazing/lib/python3.7/site-packages/cudf/core/column/string.py in normalize_characters(self, do_lower, **kwargs)
   3683         """
   3684         return self._return_or_inplace(
-> 3685             cpp_normalize_characters(self._column, do_lower), **kwargs
   3686         )
   3687 

cudf/_lib/nvtext/normalize.pyx in cudf._lib.nvtext.normalize.normalize_characters()

RuntimeError: for_each: failed to synchronize: cudaErrorIllegalAddress: an illegal memory access was encountered
```

Applying the function to `C` and `D` works. Let me know anything else I can help.

**Comment by davidwendt** [comment]
This should be fixed now with the latest branch-0.16 code now that #5919 has been merged.
```
>>> import cudf
>>> cdf = cudf.DataFrame({"A":[None, None,None],"B":[None,"1",None],"C":["1","1","1"], "D":["1","1",None]})
>>> cdf["A"].astype(str).str.normalize_characters()
0    <NA>
1    <NA>
2    <NA>
Name: A, dtype: object
>>> cdf["B"].astype(str).str.normalize_characters()
0    <NA>
1       1
2    <NA>
Name: B, dtype: object
>>> cdf["C"].astype(str).str.normalize_characters()
0    1
1    1
2    1
Name: C, dtype: object
>>> cdf["D"].astype(str).str.normalize_characters()
0       1
1       1
2    <NA>
Name: D, dtype: object

```
